src/NUnitFramework/framework/Internal/TestFilter.cs

```diff
@@ -123,11 +124,15 @@ public bool MatchParent(ITest test)
         /// <returns>True if at least one descendant matches the filter criteria</returns>
         protected virtual bool MatchDescendant(ITest test)
         {
-            if (test.Tests == null)
+            var tests = test.Tests;
+            if (tests == null)
                 return false;
 
-            foreach (ITest child in test.Tests)
+            // Use for-loop to avoid allocating the enumerator
+            int count = tests.Count;
+            for (var index = 0; index < count; index++)
```

Comment:
Another interesting performance change that I wouldn't have thought about. I'll take your word that you benchmarked each of these changes 😄